sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherClientBuilder.java

```diff
@@ -102,19 +100,12 @@ public EventGridPublisherClientBuilder() {
      * @return a publisher client with asynchronous publishing methods.
      */
     private <T> EventGridPublisherAsyncClient<T> buildAsyncClient(Class<T> eventClass) {
-        String hostname;
-        try {
-            hostname = new URL(Objects.requireNonNull(endpoint, "endpoint cannot be null")).getHost();
-        } catch (MalformedURLException e) {
-            throw logger.logExceptionAsError(new IllegalArgumentException("Cannot parse endpoint"));
-        }
```

Comment:
I think we should keep this validation. We do this in all other client builders too:
- [ServiceBusAdministrationClientBuilder](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClientBuilder.java#L159-L167)
- [TextAnalyticsClientBuilder](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java#L239-L253)
- [ConfigurationClientBuilder](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java#L216-L222)
- [SearchClientBuilder](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClientBuilder.java#L177-L183)

Reply:
Good catch - this is missing in .NET also.

Reply:
Added the url null check format validation to method `endpoint()` and added null check to `buildAsyncClient`.